bitrise.yml

```diff
@@ -34,18 +34,50 @@ workflows:
             export APPLICATION_VERSION_NAME="$(date +%s)"
             echo "Generated version_name: $APPLICATION_VERSION_NAME"
             envman add --key APPLICATION_VERSION_NAME --value "$APPLICATION_VERSION_NAME"
-    - file-downloader:
+    - activate-ssh-key@4: {}
+    - git-clone@8: {}
+    - script:
+        title: stuff
+        is_always_run: true
         inputs:
-        - destination: "/bitrise/src/_tmp/android_file"
-        - source: "$BITRISEIO_android_file_URL"
-    - git::https://github.com/DataDog/synthetics-test-automation-bitrise-step-upload-application.git:
-    # TODO: go back to using `path::./`
+        - content: |-
+            #!/bin/bash
+            pwd
+            ls -lha
+            ls ../ -lha
+    - path::./_tmp:
+        title: Upload Android application
         inputs:
         - api_key: $DATADOG_API_KEY
         - app_key: $DATADOG_APP_KEY
-        - mobile_application_version_file_path: "/bitrise/src/_tmp/android_file"
+        - mobile_application_version_file_path: "/bitrise/src/_tmp/Miniapp16-02-2024.apk"
         - mobile_application_id: "7b54b434-59ea-4ef7-868b-12c9180842f7"
         - version_name: "$APPLICATION_VERSION_NAME"
+        - latest: "true"
+    - datadog-mobile-app-run-tests@1:
+        title: Run Android tests
+        inputs:
+        - api_key: $DATADOG_API_KEY
+        - app_key: $DATADOG_APP_KEY
+        - public_ids: "rxd-vdv-qce"
+        - fail_on_critical_errors: "true"
+    - path::./_tmp:
```

Comment:
Instead of trying to reinvent the wheel, you can search for what people already did [in that big ol' archive of code that is GitHub](https://github.com/search?q=%22git-clone%22+AND+%22path%3A%3A.%2F%22&type=code), and it looks like the [best solution is to do this](https://github.com/wpinaud/steps-xcode-archive/blob/6dbc53d834942da667c44689779b44fae38592f5/bitrise.yml#L66):

- Keep `change-workdir` to `_tmp` because this is an advice from Bitirse for better step testing
- Use `clone_into_dir: ./` with `git-clone`. Note that `./` will refer to `./_tmp` since we just `cd`'d into it in the previous step
- Then use `path::./` as it looks better

Reply:
wait i don't understand how this fixes the issue 🤔 
we need `path::./_tmp` because by default `path::./` searches for `bitrise/src/step.yml` however with the git-clone step, even without the `clone_into_dir` thing, clones the repo into `_tmp`, so the step.yml is in fact in `bitrise/src/_tmp/step.yml`.

Reply:
> because by default `path::./` searches for `bitrise/src/step.yml`

Yes, but only because by default, `.` is `bitrise/src`.

Since we change the workdir to be `./_tmp`, `.` becomes `bitrise/src/_tmp`. So `path::./` should start searching for `bitrise/src/_tmp/step.yml`.

Reply:
> Since we change the workdir to be ./_tmp, . becomes bitrise/src/_tmp

I think it doesn't work like that sadly 😅 
after the `change-workdir` step if i do `pwd` i'm already in `bitrise/src/_tmp` and the git clone does correctly then clone in there,  but the `path::./` is still looking in `bitrise/src`

I think the `clone_into_dir` is just basically a path to where to clone to and defaults to `$BITRISE_SOURCE_DIR, which might be the env var that we actually want to change, but not sure i want to go and change internal bitrise envs 🤔

Reply:
We talked more and in the end decided to get rid of the `change-workdir` step as `path::./` seems a lot cleaner